### PR TITLE
various comms-GC-related cleanups

### PR DESCRIPTION
**Pull request by warner**
These commits smooth the way for the upcoming GC-over-comms branch. Mostly refactorings, although the `processMaybeFree` rearrangement has the potential to slightly modify the behavior (hopefully in a good way).

refs #3306

### CODE REVIEW

File: packages/SwingSet/src/vats/comms/state.js [hunk 1]
```diff
@@ -220,27 +220,36 @@ export function makeState(syscall, identifierBase = 0) {
...
+    while (maybeFree.size) {
+      const lrefs = Array.from(maybeFree.values());
+      lrefs.sort();
+      maybeFree.clear();
+      for (const lref of lrefs) {
+        const { type } = parseLocalSlot(lref);
+        if (type === 'promise' && enableLocalPromiseGC) {
+          const s = store.get(`${lref}.refCount`);
+          if (s) {
+            const refCount = Number(s);
```
Comment (dckc) [inline packages/SwingSet/src/vats/comms/state.js]: any particular reason why a `Number` object rather than a `number` primitive?

Comment (dckc) [reply packages/SwingSet/src/vats/comms/state.js]:
```suggestion
            const refCount = parseInt(s, 10);
```

File: packages/SwingSet/src/vats/comms/dispatch.js [hunk 2]
```diff
@@ -141,70 +141,59 @@ export function buildCommsDispatch(
...
-        return;
+        resolveFromKernel(resolutions);
+        break;
       }
       case 'dropExports': {
         const [vrefs] = args;
-        dropExports(vrefs);
-        return;
+        assert(Array.isArray(vrefs));
+        vrefs.map(vref => insistVatType('object', vref));
```
Comment (dckc) [inline packages/SwingSet/src/vats/comms/dispatch.js:161]: when not using the return value, `forEach` seems more appropriate than `map`

**Review by dckc — COMMENTED** [review]
I haven't studied this part of the code enough to be confident that these changes preserve correctness; I just looked over the diffs and made a couple editorial comments.

File: packages/SwingSet/src/vats/comms/state.js (hunk 1 above)
Comment (warner) [reply packages/SwingSet/src/vats/comms/state.js]: Good point, all I want is a non-negative integer from a string. We're being inconsistent here w.r.t. BigInt vs Number. I'll go with the `parseInt` for now, thanks.

File: packages/SwingSet/src/vats/comms/dispatch.js (hunk 2 above)
Comment (FUDCo) [reply packages/SwingSet/src/vats/comms/dispatch.js:161]: There's lots of JS lore to the effect of "never use `forEach`", but I'm not sure the extent to which this is superstition, or a nana's pot roast thing, or whether there's some subtlety I haven't followed.

**Review by FUDCo — APPROVED** [review]
This all seems fine.

**Comment by warner** [comment]
@FUDCo I added two more commits that warrant your attention, as they change the one-sided c-list retirement code we discussed many moons ago.

**Review by FUDCo — APPROVED** [review]
Es ist gut.

**Pull request merged**